src/snowflake/snowpark/session.py

```diff
@@ -558,11 +560,17 @@ def __init__(
                 _PYTHON_SNOWPARK_USE_SQL_SIMPLIFIER_STRING, True
             )
         )
-        self._cte_optimization_enabled: bool = (
+        cte_optimization_enabled_version = (
```

Comment:
Why only the parameters for cte optimization and temp table auto cleanup depend on snowpack's version but not the other params (e.g., describe queries, large query breakdown, etc)?

Reply:
yes other parameters will also be changed in this way later!